### PR TITLE
Add getNetworkStats method on apiProvider

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: erdjs/src/stats.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    static getDefault(): Stats {
+        if (!Stats.default) {
+            Stats.default = new Stats();
+        }
+
+        return Stats.default;
+    }
+
+    /**
+     * Synchronizes a configuration object by querying the Network, through a {@link IProvider}.
```
Comment (andreibancioiu) [inline erdjs/src/stats.ts:66]: `a configuration object` -> `a stats object`?

File: erdjs/src/stats.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        this.RefreshRate = 0;
+        this.Epoch = 0;
+        this.RoundsPassed = 0;
+        this.RoundsPerEpoch = 0;
+    }
+
+    /**
+     * Gets the default configuration object (think of the Singleton pattern).
+     */
+    static getDefault(): Stats {
```
Comment (andreibancioiu) [inline erdjs/src/stats.ts:57]: The `NetworkConfig` object is used when creating `Transaction` objects, thus we added this `getDefault()`. Do we need this for `Stats` as well?

File: erdjs/src/stats.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        this.Accounts = 0;
+        this.Transactions = 0;
+        this.RefreshRate = 0;
+        this.Epoch = 0;
+        this.RoundsPassed = 0;
+        this.RoundsPerEpoch = 0;
+    }
+
+    /**
+     * Gets the default configuration object (think of the Singleton pattern).
```
Comment (andreibancioiu) [inline erdjs/src/stats.ts:55]: "the default `Stats` object"?

File: erdjs/src/stats.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+     * Synchronizes a configuration object by querying the Network, through a {@link IProvider}.
+     * @param provider The provider to use
+     */
+    async sync(provider: IApiProvider): Promise<void> {
+        let fresh: Stats = await provider.getNetworkStats();
+        Object.assign(this, fresh);
+    }
+
+    /**
+     * Constructs a configuration object from a HTTP response (as returned by the provider).
```
Comment (andreibancioiu) [inline erdjs/src/stats.ts:75]: "constructs a `Stats` object" etc.

**Review by andreibancioiu — CHANGES_REQUESTED** [review]

File: erdjs/src/stats.ts (hunk 1 above)
Comment (danielailie) [reply erdjs/src/stats.ts]: Done

File: erdjs/src/stats.ts (hunk 2 above)
Comment (danielailie) [reply erdjs/src/stats.ts]: Nope, I just removed the getDefault Function

File: erdjs/src/stats.ts (hunk 3 above)
Comment (danielailie) [reply erdjs/src/stats.ts]: done

File: erdjs/src/stats.ts (hunk 4 above)
Comment (danielailie) [reply erdjs/src/stats.ts]: done

**Review by andreibancioiu — DISMISSED** [review]

**Review by andreibancioiu — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Pull request merged**